crates/astria-sequencer/src/asset/query.rs

```diff
@@ -96,18 +99,16 @@ fn preprocess_request(params: &[(String, String)]) -> anyhow::Result<asset::Id,
             ..response::Query::default()
         });
     };
-    let asset_id = hex::decode(asset_id)
+    let asset = <[u8; 32]>::from_hex(asset_id)
```

Comment:
I am torn on this one. The request still has the form `asset/denom/<hex>`. So `asset/denom/ibc/<hex>` isn't allowed, and neither is `asset/denom/port/channel/base`.